pkgs/servers/typedb/default.nix

```
@@ -0,0 +1,97 @@
+{ stdenv, lib, openjdk, system, fetchzip, fetchurl, server-data ? "~/.typedb_home", server-port ? "1729", server-logs ? "~/.typedb_home/logs" }:
+let
+  typedbVersion = "2.1.1";
+  systems = {
+    x86_64-linux = {
+      src = fetchurl {
+        url = "https://github.com/vaticle/typedb/releases/download/2.1.1/typedb-all-linux-2.1.1.tar.gz";
+        sha256 = "00wvkiahmrys32ckp6r59bbrzfd7jzsphniyjdj4p9bh7k339gn2";
+      };
+      dir = "typedb-all-linux-${typedbVersion}";
+    };
+    x86_64-darwin = {
+      src = fetchzip {
+        url = "https://github.com/vaticle/typedb/releases/download/2.1.1/typedb-all-mac-2.1.1.zip";
+        sha256 = "16hlfy6kh2rnvcralz206q13mghb0rv8wazpg6q3h324p5rdys54";
+      };
+      dir = "typedb-all-mac-${typedbVersion}";
+    };
+    x86_64-windows = {
+      src = fetchzip {
+        url = "https://github.com/vaticle/typedb/releases/download/2.1.1/typedb-all-windows-2.1.1.zip";
+        sha256 = "0vd66gfshkg697z07nhy957mwqzlli4r4pmn67hx58n9mkg024kq";
+      };
+      dir = "typedb-all-windows-${typedbVersion}";
+    };
+  };
+
+  currentSystem = systems.${system} or throw "unsupported system ${system}";
```

My bad. I always get this wrong. `x . y or z` is tighter than function application.

```suggestion
  currentSystem = systems.${system} or (throw "unsupported system ${system}");
```